pkg/api/meta/builds.go

```diff
@@ -0,0 +1,55 @@
+package meta
+
+import (
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+)
+
+type buildSpecMutator struct {
+	spec *buildapi.CommonSpec
+	path *field.Path
+}
+
+func (m *buildSpecMutator) Mutate(fn ImageReferenceMutateFunc) field.ErrorList {
+	var errs field.ErrorList
+	for i, image := range m.spec.Source.Images {
+		if image.From.Kind != "DockerImage" {
```

Comment:
When I'm using an istag, at what point does the resolution down to a particular image happen?  Seems like we'd need to control that resolution and have it check for denials.